pallets/asset/src/checkpoint/mod.rs

```diff
@@ -124,7 +124,7 @@ impl From<Moment> for ScheduleSpec {
     }
 }
 
-storage_migration_ver!(1);
+storage_migration_ver!(2);
```

Comment:
I have seen enums used here instead of an int.
e.g
```
enum Releases {
    V1_0_0Ancient,
    V2_0_0,
    V3_0_0,
    V4_0_0,
    V5_0_0,
    V6_0_0,
}
```

Any thoughts?

Reply:
I personally prefer ints but substrate uses enums so we have a mix where our pallets use ints while substrate inherited pallets like staking use enums.

Reply:
So we had a tech discussion on this and decided to go with a mix of integers and compile time checks defined by `storage_migrate_ver!(MAX_VERSION)` and `storage_migrate_on!(..., match_version, ...)`. Let's execute on the outcome... :)